crates/client/gateway/src/server/handler.rs

```diff
@@ -0,0 +1,307 @@
+use std::sync::Arc;
+
+use hyper::{body, Body, Request, Response};
+use mc_db::MadaraBackend;
+use mc_rpc::providers::AddTransactionProvider;
+use mp_block::{BlockId, BlockTag, MadaraBlock, MadaraPendingBlock};
+use mp_class::ContractClass;
+use mp_gateway::{
+    block::{BlockProvider, BlockStatus, PendingBlockProvider},
+    state_update::{PendingStateUpdateProvider, StateUpdateProvider},
+};
+use serde_json::json;
+use starknet_core::types::{
+    BroadcastedDeclareTransaction, BroadcastedDeployAccountTransaction, BroadcastedInvokeTransaction,
+    BroadcastedTransaction,
+};
+use starknet_types_core::felt::Felt;
+
+use crate::error::{StarknetError, StarknetErrorCode};
+
+use super::{
+    error::{GatewayError, ResultExt},
+    helpers::{
+        block_id_from_params, create_json_response, create_response_with_json_body, get_params_from_request,
+        include_block_params,
+    },
+};
+
+pub async fn handle_get_block(req: Request<Body>, backend: Arc<MadaraBackend>) -> Response<Body> {
+    let params = get_params_from_request(&req);
+    let block_id = match block_id_from_params(&params) {
+        Ok(block_id) => block_id,
+        // Return the error response if the request is malformed
+        Err(e) => return e.into(),
+    };
+
+    let block = match backend
+        .get_block(&block_id)
+        .or_internal_server_error("get_block")
```

Comment:
this should be human readable, like the rpc errors
```rs

    let class_hash = starknet
        .backend
        .get_contract_class_hash_at(&resolved_block_id, &contract_address)
        .or_internal_server_error("Error getting contract class hash at")?
        .ok_or(StarknetRpcApiError::ContractNotFound)?
```

Reply:
same for all the others

Reply:
also, add as much context as possible if possible
example 
```rs
        .or_else_internal_server_error(|| format!("Error getting contract class hash at {block_id}"))?
```
by adding an or_else variant too - like you would with option.ok_or(Error::Thing) vs options.ok_or_else(fn once)